### PR TITLE
Fix/185 xss regression

### PR DESCRIPTION
**Pull request by der**
- revert incorrect xss clean for login close button
- trap cases of non-existent external references

Addresses UKGovLD/registry-core#/185 comment https://github.com/UKGovLD/registry-core/issues/185#issuecomment-1545866128

Despite the complexity of the diff the actual change to ldregistry/templates/main/_fetch_item.vm is to insert an extra condition at lines 60-62 then indent the remainder of the code in the else branch closing it line 100.

### CODE REVIEW

**Review by simonoakesepimorphics — APPROVED** [review]

**Review by simonoakesepimorphics — APPROVED** [review]

File: ldregistry/templates/main.vm [hunk 1]
```diff
@@ -2,7 +2,7 @@
 ##
 ## Input parameters:
 ##   $uri     - the requested URI, relative to the registry root (generated by registry-core)
-##   $entity  - the URI of a external entity to be displayed, passed as a query parameter, optional
+##   $entity  - no longer supported
 ##   $_browse - flag (query parameter) to indicate user has selected "browse"  to override landing page, optional
 
 #set( $isLandingPage = $uri.isEmpty() && !$entity && !${_browse} )
```
Comment (mike-gormley) [inline ldregistry/templates/main.vm:8]:
```suggestion
#set( $isLandingPage = $uri.isEmpty() && !${_browse} )
```
The inclusion of this here means entity is not being ignored completely, even if entity is an empty string.

**Comment by der** [comment]
> Within the _not-found.vim template, then the ‘not found’ URI is repeated back in the response payload
> ldregistry/templates/main/_not-found.vm

Good point, I've suppressed that completely now.

**Review by mo-marqh — APPROVED** [review]

File: ldregistry/templates/main/_not-found.vm [hunk 2]
```diff
@@ -1,6 +1,6 @@
   <div class="jumbotron">
     <h1>$msg['error.notFound.heading']</h1>
     <p>
-      $msg.get('error.notFound.note', $target, $lib.escapeHtml($target))
+      $msg.get('error.notFound.note')
```
Comment (mike-gormley) [inline ldregistry/templates/main/_not-found.vm:4]: Perhaps https://github.com/UKGovLD/registry-config-base/blob/master/ldregistry/config/language/messages/en.properties#L80 should also be updated to remove the placeholder to something like `error.notFound.note=Sorry, the requested URI was not found on this site.`?

Comment (der) [reply ldregistry/templates/main/_not-found.vm:4]: Thanks Mike, yes, now done. I had that on a list to check on but juggling too many things at once.

File: ldregistry/config/language/messages/en.properties [hunk 3]
```diff
@@ -63,7 +63,7 @@ checkUri.label=Check URI
 checkUri.heading=Check whether a URI is registered
 checkUri.button=Check
 checkUri.uri=URI to check
-checkUri.notFound=<strong>Not found.</strong> URI <{0}> does not seem to be registered.
+checkUri.notFound=<strong>Not found.</strong> =Sorry, the requested URI was not found on this site.
```
Comment (mike-gormley) [inline ldregistry/config/language/messages/en.properties:66]:
```suggestion
checkUri.notFound=<strong>Not found.</strong> Sorry, the requested URI was not found on this site.
```
An extra `=` had sneaked in there. Line 80 needs adjusting too.

**Comment by der** [comment]
HI @mo-marqh 

> i think it would be good to take this progress and merge, then address any potential follow up facets in a separate PR
> is this suitable to be merged now?

I'm afraid not. We await more information from @sgrellet in response to https://github.com/UKGovLD/registry-core/issues/185#issuecomment-1550312656

It seems that the external entity rendering feature might be in production use there which would call into question the whole strategy.

**Review by mike-gormley — APPROVED** [review]

**Pull request merged**